src/sentry/auth/authenticators/u2f.py

```diff
@@ -57,11 +79,17 @@ def is_available(cls):
         url_prefix = options.get("system.url-prefix")
         return url_prefix and url_prefix.startswith("https://")
 
-    def _create_credential_object(self, registeredKey):
-        return base.AttestedCredentialData.from_ctap1(
-            websafe_decode(registeredKey["keyHandle"]),
-            websafe_decode(registeredKey["publicKey"]),
-        )
+    def _get_kept_devices(self, key):
+        # return a list of devices that doesn't match the key
+        return [
+            x
+            for x in self.config.get("devices") or ()
+            if (
+                decode_credential_id(x) != key
+                if type(x["binding"]) == AuthenticatorData
+                else x["binding"]["keyHandle"] != key
+            )
```

Comment:
could this be rewritten so that it's not a nested if statement?

Reply:
i can think of doing it this way
`devices = []
        for device in self.config.get("devices", ()):
            if (type(device["binding"]) == AuthenticatorData and decode_credential_id(device) != key) or device["binding"]["keyHandle"] != key:
                devices.append(device)

        return devices`

Reply:
nvm the above doesn't seem to work